### PR TITLE
ObjC - Adds class factory method to generated ProtoService classes

### LINKED ISSUE #3472: ObjC - Add factory method to ProtoService generated classes
Currently, the only way to create a new instance of a ProtoService class is to use:

    - (instancetype)initWithHost:(NSString *)host NS_DESIGNATED_INITIALIZER;

I have a wrapper class that initializes a ProtoService ivar as part of its own init sequence. The only way for me to mock the ProtoService (for unit testing) is to create an abstract factory class that exposes a factory method to create this ProtoService. A more idiomatic and elegant solution is to expose a factory method on the generated ProtoService directly, so instead of:

    [[MyProtoService alloc] initWithHost:@""]

... it becomes:

    [MyProtoService createServiceWithHost:@""]

This will allow frameworks like OCMock to mock the class method `createServiceWithHost:` without a need for the caller to create their own abstract factory class.

### PR DESCRIPTION
**Pull request by nkibler**
Fixes #3472.

### CODE REVIEW

File: src/compiler/objective_c_generator.cc [hunk 1]
```diff
@@ -239,6 +240,10 @@ ::grpc::string GetSource(const ServiceDescriptor *service) {
     printer.Print("                 packageName:(NSString *)packageName\n");
     printer.Print("                 serviceName:(NSString *)serviceName {\n");
     printer.Print("  return [self initWithHost:host];\n");
+    printer.Print("}\n\n");
+    printer.Print("// Class factory method\n");
```
Comment (jcanizales) [inline src/compiler/objective_c_generator.cc]: No need for the comment, as it's pretty straightforward :)

**Comment by jcanizales** [comment]
Unlike with Perforce, you don't need to be merging master into your branch to keep the gods happy :) Git merge is smart enough to solve conflicts automatically whenever it's safe.

**Comment by jcanizales** [comment]
Would you changing https://github.com/grpc/grpc/blob/master/src/objective-c/tests/InteropTests.m#L92 to use this and then running, from the root of your local repo:

```bash
./tools/run_tests/run_tests.py -l objc
```

**Comment by nkibler** [comment]
Tests pass with latest fixes to C core.
screen shot 2015-09-29 at 11 22 25 am

**Pull request merged**